tools/oeedger8r/Emitter.ml

```diff
@@ -967,274 +472,861 @@ let warn_size_and_count_params (fd : func_decl) =
       | _ -> () )
     fd.plist
 
-(** Validate Open Enclave supported EDL features. *)
-let validate_oe_support (ec : enclave_content) (ep : edger8r_params) =
-  (* check supported options *)
-  if ep.use_prefix then
-    failwithf "--use_prefix option is not supported by oeedger8r." ;
-  List.iter
-    (fun f ->
-      if f.tf_is_priv then
-        failwithf
-          "Function '%s': 'private' specifier is not supported by oeedger8r"
-          f.tf_fdecl.fname ;
-      if f.tf_is_switchless then
-        failwithf
-          "Function '%s': switchless ecalls and ocalls are not yet supported \
-           by Open Enclave SDK."
-          f.tf_fdecl.fname )
-    ec.tfunc_decls ;
-  List.iter
-    (fun f ->
-      ( if f.uf_fattr.fa_convention <> CC_NONE then
-        let cconv_str = get_call_conv_str f.uf_fattr.fa_convention in
-        printf
-          "Warning: Function '%s': Calling convention '%s' for ocalls is not \
-           supported by oeedger8r.\n"
-          f.uf_fdecl.fname cconv_str ) ;
-      if f.uf_fattr.fa_dllimport then
-        failwithf "Function '%s': dllimport is not supported by oeedger8r."
-          f.uf_fdecl.fname ;
-      if f.uf_allow_list != [] then
-        printf
-          "Warning: Function '%s': Reentrant ocalls are not supported by Open \
-           Enclave. Allow list ignored.\n"
-          f.uf_fdecl.fname ;
-      if f.uf_is_switchless then
-        failwithf
-          "Function '%s': switchless ecalls and ocalls are not yet supported \
-           by Open Enclave SDK."
-          f.uf_fdecl.fname )
-    ec.ufunc_decls ;
-  (* Map warning functions over trusted and untrusted function
-     declarations *)
-  let ufuncs = List.map (fun f -> f.uf_fdecl) ec.ufunc_decls in
-  let tfuncs = List.map (fun f -> f.tf_fdecl) ec.tfunc_decls in
-  let funcs = List.append ufuncs tfuncs in
-  List.iter
-    (fun f ->
-      warn_non_portable_types f ;
-      warn_signed_size_or_count_types f ;
-      warn_size_and_count_params f )
-    funcs
-
-(** Includes are emitted in [args.h]. Imported functions have already
-    been brought into function lists. *)
-let gen_t_h (ec : enclave_content) (ep : edger8r_params) =
-  let oe_gen_tfunc_prototypes (tfs : trusted_func list) =
-    if tfs <> [] then
-      List.map (fun f -> sprintf "%s;" (oe_gen_prototype f.tf_fdecl)) tfs
-    else ["/* There were no ecalls. */"]
+(** Generate the Enclave code. *)
+let gen_enclave_code (ec : enclave_content) (ep : edger8r_params) =
```

Comment:
This is now the "top-level" function, meaning we can eliminate the explicit passing of `ec` and `ep` to all the nested functions.